sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/options/ProxyInvocationHandler.java

```diff
@@ -355,10 +472,24 @@ public void serialize(PipelineOptions value, JsonGenerator jgen, SerializerProvi
         // instances that have been modified since the previous serialization.
         Map<String, Object> serializableOptions =
             Maps.<String, Object>newHashMap(handler.jsonOptions);
-        serializableOptions.putAll(filteredOptions);
+        for (Map.Entry<String, BoundValue> entry : filteredOptions.entrySet()) {
+          serializableOptions.put(entry.getKey(), entry.getValue().getValue());
+        }
+
+
         jgen.writeStartObject();
         jgen.writeFieldName("options");
         jgen.writeObject(serializableOptions);
+
+        if (handler.canPopulateDisplayData()) {
```

Comment:
May be worth failing or logging if we can't do this. If I read correctly, this would happen if we're trying to run a pipeline from options that were deserialized from JSON. That typically shouldn't happen, so maybe we should just throw?